### PR TITLE
Added default "static/<path>" route for any static files

### LINKED ISSUE #41: static directory support
Have the `/static` directory be fully accessible to clients, this is similar to how Flask handles static files.

This can be done either by having a `/static/<path>` rule be always present (possible with an option to disable it). or with middleware that checks for any requests that start with `/static` and just does a `set_static_file_info` to whatever the request url is.

ipkn#386 is somewhat related.

### PR DESCRIPTION
**Pull request by The-EDev**
allows anyone to put whatever file in a `static` folder, the files there will be fully accessible.

can be disabled via `CROW_DISABLE_STATIC_DIR` def.

closes #41

### CODE REVIEW

File: include/crow/app.h [hunk 1]
```diff
@@ -161,7 +161,16 @@ namespace crow
         ///Run the server
         void run()
         {
+#ifndef CROW_DISABLE_STATIC_DIR
+            route<crow::black_magic::get_parameter_tag("/static/<path>")>("/static/<path>")
```
Comment (mrozigor) [inline include/crow/app.h:165]: Maybe extract `static` to const and allow user to change it on compile time?

Comment (The-EDev) [reply include/crow/app.h:165]: I'll look into it

Comment (The-EDev) [reply include/crow/app.h:165]: I'm not sure how the user can modify a const on compile time, is there a way of setting a const on compile time without say macros or some cast?

Comment (mrozigor) [reply include/crow/app.h:165]: Sorry, my bad. I thought about changing this variable before merging all includes, but that's bad idea. Maybe some configuration property?

Comment (The-EDev) [reply include/crow/app.h:165]: it's possible to add a macro like `CROW_LOG_LEVEL` in `settings.h`, but right now all it takes the user to set their own static directory is to set the disable flag and create a 2 line route. I'm fine with both adding a macro or not changing anything :)

Comment (mrozigor) [reply include/crow/app.h:165]: Ok, so we can stay with your solution :)

**Review by mrozigor — DISMISSED** [review]

File: include/crow/app.h (hunk 1 above)
Comment (The-EDev) [reply include/crow/app.h:165]: oh man, I just made it work with macros 😅 XD

**Review by mrozigor — APPROVED** [review]

**Pull request merged**